orderer/common/multichannel/registrar.go

```diff
@@ -495,6 +495,8 @@ func (r *Registrar) JoinChannel(channelID string, configBlock *cb.Block, isAppCh
 // RemoveChannel instructs the orderer to remove a channel.
 // Depending on the removeStorage parameter, the storage resources are either removed or archived.
 func (r *Registrar) RemoveChannel(channelID string, removeStorage bool) error {
-	//TODO
+	if r.SystemChannelID() != "" && channelID != r.SystemChannelID() {
+		return errors.Errorf("cannot remove application channel %s because the system channel exists", channelID)
+	}
```

Comment:
The REST API checks for 2 explicit error codes:
```go
	case types.ErrSystemChannelExists:
		h.sendResponseNotAllowed(resp, errors.Wrap(err, "cannot remove"), http.MethodGet)
	case types.ErrChannelNotExist:
		h.sendResponseJsonError(resp, http.StatusNotFound, errors.Wrap(err, "cannot remove"))
```
see [link](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/orderer/common/channelparticipation/restapi.go#L270).

So you need to return a specific error in order for the REST response error code & headers to be correct.

Reply:
Thanks for that pointing that out. Not sure how I missed it. :) Done.